### ISSUE TITLE
Status Codes

### ISSUE THREAD

**Issue by cjenaro** [sure-thing/presta#77]
Hi! Just found the project and I believe it has great potential.

I've been messing around with forms in it and I cannot redirect or get the form to post to the same route, e.g:

```js
import { html } from "presta/html";

export const route = "/";


function post(props) {
  // do form stuff and then redirect
  return {
    statusCode: 303,
    headers: {
      Location: "/",
    },
  };
}

export async function handler(props) {
 // if is post then run the post function
  if (props.lambda.event.httpMethod.toLowerCase() === "post") {
    return post(props.lambda);
  }
  
  return html({
    head: {
      title: "Title",
    },
    body: `<h1>hello!</h1>
    
      <form method="post">
        <input type="text" name="name" />
        <button type="submit">Go</button>
      </form>
    `,
  });
}
```
Hopefully I explained myself correcty, I can't get the statusCode on the response to work, moreover, when entering a 404 url I get the 404 page but the statusCode is 200, which leads me to believe that there is a 200 hardcoded somewhere, not sure where, when the response starts with 3 and there is a Location header, the browser should redirect automatically.

**Comment by estrattonbailey** [comment]
Hey @cjenaro! Thanks for the issue, I really appreciate it.

So glad you tried a form — it's been on my list, and I'm excited to be able to support what many might consider an antiquated pattern of native form submissions.

You were right, it was returning a 200. It was due to [a typo](https://github.com/sure-thing/presta/commit/ebddbfe8ea679717da580d8f4a182b81dafcefbd#diff-90b8c40654346a0f90429a121bea2ca1f98e8678a1ad52764eb95aaae1f46159L193) 😛 However, in testing this, I realized we needed a better body parsing solution, and a couple other things. I also made an [example](https://github.com/sure-thing/presta-examples/tree/master/form) of native form submissions for others to see.

The latest release `0.31.13` should work for you to start, but I'm sure there are further improvements we could make. If you're interested at all in contributing we'd be glad to have you! Thanks again 👍
